### PR TITLE
added uv support and ruff, pytest as dev. dependencies

### PR DESCRIPTION
**Pull request by SermetPekin**
added uv support. 
now it can be used with uv environment package 
```bash
pip install uv -U 
uv pip install .
uv pip install -e . 

```

added ruff formatter package as a dev. dependency. 
```bash
uv pip install -e .
ruff check . # checks for formatting, unused imports etc. 
pytest -x -v # tests are successful
```

### CODE REVIEW

File: .idea/.gitignore
Comment (russellgoyder) [inline .idea/.gitignore:1]: Is this for PyCharm?

**Review by russellgoyder — APPROVED** [review]
Thank you @SermetPekin, this is great - uv was on my list 🙂

**Comment by SermetPekin** [comment]
.idea file was created by pycharm and I think it should be ignored on .gitignore file. I accidentally pushed that sorry.

**Comment by russellgoyder** [comment]
Ah - I already merged it! We can tidy it up later.

**Pull request merged**